px4_ros2_cpp/include/px4_ros2/mission/mission_executor.hpp

```diff
@@ -425,6 +444,35 @@ class ActionHandler
     _mission_executor.setTrajectoryOptions(options);
   }
 
+  /**
+   * @brief Sets the activity info for the mission executor.
+   *        This method forwards the provided activity string to the underlying mission executor, 
+   *        making it externally accessible. This string provides a concise description of the current action.
+   *
+   * @param activityInfo The string describing the current action.
+   */
+  void setActvityInfo(const std::string& activityInfo) 
+  {
+    if (!_valid) {
+      RCLCPP_WARN(_mission_executor._node.get_logger(), "ActionHandler is not valid anymore");
+      return;
+    }
+    _mission_executor.setActivityInfo(activityInfo);
+  }
+
+  /**
+   * @brief Clears the activity information.
+   *        This method sets the activity info of the mission executor to `std::nullopt`. 
+   *        It should be called after an action is completed to ensure the state is properly reset.
```

Comment:
Let's call this automatically when switching to the next waypoint, or when the mission is deactivated or aborted.